### PR TITLE
Check if commit was signed by an anonymous email

### LINKED ISSUE #118: Refuse DCO for users.noreply.github.com
## Expected Behaviour

The DCO requires a valid email address and not an anonymous entity.

## Current Behaviour

Some users put `user@users.noreply.github.com` to obstinate their real email address rendering it useless for the purposes it is required.

## Possible Solution

Add the `no-dco` label and a comment about the use of an anonymous email.

## Steps to Reproduce (for bugs)

1. Sign off a commit as "Name <user@users.noreply.github.com>" and see Derek accept that

### PR DESCRIPTION
**Pull request by matipan**
Signed-off-by: Matias Pan <matias.pan26@gmail.com>

This PR adds a check for when commits have an anonymous email address.

## Description

If at least one of the commits was signed with an email address that ends in `noreply.github.com` then we reply with a different comment that asks for the commit to be properly signed. This is comment that we are using(it can be changed :smile: ): 
> Thank you for your contribution. It seems that one or more of your commits have an anonymous email address. Please consider signing your commits with a valid email address. Please see our [contributing guide](https://github.com/matipan/test-derek/blob/master/CONTRIBUTING.md).

## Motivation and Context

- [x] I have raised an issue to propose this change (Fixes #118)

## How Has This Been Tested?

Unit tests for new functionality
I built derek at `matipan/derek:anonymousCheck`, deployed it on my personal cluster and tested it using the following repository: [matipan/test-derek](https://github.com/matipan/test-derek)
Derek in action:
derek-dco-invalid-email
Existing behavior was not changed:
derek-dco-unsigned
derek-dco-valid

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/alexellis/derek/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: handler/pullRequestHandler_test.go [hunk 1]
```diff
@@ -124,3 +124,81 @@ func Test_hasDescription(t *testing.T) {
...
+	}
+}
+
+func Test_hasAnonymousSign(t *testing.T) {
+	var anonymousSignOpts = []struct {
+		title    string
+		commits  []*github.RepositoryCommit
+		expected bool
+	}{
+		{
```
Comment (alexellis) [inline handler/pullRequestHandler_test.go]: There is a third use-case here, where there isn't any sign-off message. Let's add it?

File: handler/pullRequestHandler.go [hunk 2]
```diff
@@ -45,56 +46,60 @@ func HandlePullRequest(req types.PullRequestOuter, contributingURL string, confi
...
+
+	var body string
+	if anonymousSign {
+		body =
+			`Thank you for your contribution. It seems that one or more of your commits have an anonymous email address. Please consider signing your commits with a valid email address. Please see our [contributing guide](` + contributingURL + `).`
 	} else {
-		fmt.Println("Things look OK right now.")
-		issue, res, labelErr := client.Issues.Get(ctx, req.Repository.Owner.Login, req.Repository.Name, req.PullRequest.Number)
+		body =
+			`Thank you for your contribution. I've just checked and your commit doesn't appear to be signed-off. That's something we need before your Pull Request can be merged. Please see our [contributing guide](` + contributingURL + `).
```
Comment (alexellis) [inline handler/pullRequestHandler.go]: For this and the other message please extract a function to generate the message. Nothing fancy, one function per message.

Comment (alexellis) [reply handler/pullRequestHandler.go]: It will help with readability and we can also test it.

File: handler/pullRequestHandler_test.go (hunk 1 above)
Comment (matipan) [reply handler/pullRequestHandler_test.go]: You are right, I'll add it

File: handler/pullRequestHandler.go (hunk 2 above)
Comment (matipan) [reply handler/pullRequestHandler.go]: No problem :+1:

**Comment by matipan** [comment]
@alexellis I updated with the requested changes. 
To clarify how `hasAnonymous` behaves. The only time we consider a commit to be anonymous is when the email address ends in `noreply.github.com`. If there is no `Signed-off-by` in the message the function returns true. We are still using the `hasUnsigned` function to check for that particular case.
What do you think about that approach?

**Comment by alexellis** [comment]
Your work so far is looking 💯, very good workflow and testing. Thank you.

Please could you see this note on commit messages and edit your current message?

https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md#commit-messages

Alex

**Comment by matipan** [comment]
Definitely. I'll fix the last commit asap, thanks for the heads up :+1:

**Comment by matipan** [comment]
@alexellis I just updated the commit message to follow the guidelines :+1:

**Comment by rgee0** [comment]
Hello @matipan - can we get the description updated so that it includes a GitHub closure keyword, please?  More here: https://help.github.com/en/articles/closing-issues-using-keywords

Convention is to add it in this line:

* I have raised an issue to propose this change (Fixes #120)

**Comment by matipan** [comment]
No problem @rgee0. I just updated it, thanks!

**Comment by alexellis** [comment]
Thank you @matipan I owe you Derek swag :-)

**Pull request merged**